plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -382,6 +382,116 @@ HAVING
+DECLARE @currIntervalStartDate datetime = DATEADD(second, @currIntervalStartTimestamp % @secondsInDay, DATEADD(day, @currIntervalStartTimestamp / @secondsInDay, 0));
+
+`
+
+const sqlAzureDBQueryStoreRuntimeStatistics = sqlAzureDBPartQueryPeriod + `
+SET DEADLOCK_PRIORITY -10;
+IF SERVERPROPERTY('EngineEdition') = 5  -- Is this Azure SQL DB?
+BEGIN
+SELECT
+	'sqlserver_azuredb_querystore_runtime_stats' AS [measurement],
+	CONVERT(varchar(36), NEWID()) AS [measurement_id],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(datetime, rsi.start_time, 1) AS interval_start_time,
+	CONVERT(datetime, rsi.end_time, 1) AS interval_end_time,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1)), CONVERT(datetime, rsi.start_time, 1))) * @magicNumberSeconds AS interval_start_time_d,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1)), CONVERT(datetime, rsi.end_time, 1))) * @magicNumberSeconds AS interval_end_time_d,
+	q.query_id,
+	CONVERT(nvarchar(max), q.query_hash, 1) as query_hash,
+	p.plan_id,
+	CONVERT(nvarchar(max), q.last_compile_batch_sql_handle, 1) as last_compile_batch_sql_handle,
+	CONVERT(nvarchar(max), p.query_plan_hash, 1) as query_plan_hash,
+	rs.max_logical_io_reads as max_logical_io_reads,
+	rs.avg_logical_io_reads as avg_logical_io_reads,
+	rs.avg_logical_io_reads as logical_io_reads,
+	rs.max_physical_io_reads as max_physical_io_reads,
+	rs.avg_physical_io_reads as avg_physical_io_reads,
+	rs.max_logical_io_writes as max_logical_io_writes,
+	rs.avg_logical_io_writes as avg_logical_io_writes,
+	rs.execution_type,
+	rs.count_executions as count_executions,
+	rs.max_cpu_time as max_cpu_time,
+	rs.avg_cpu_time as avg_cpu_time,
+	rs.max_dop as max_dop,
+	rs.avg_dop as avg_dop,
+	rs.max_rowcount as max_rowcount,
+	rs.avg_rowcount as avg_rowcount,
+	rs.max_query_max_used_memory AS max_query_max_used_memory,
+	rs.avg_query_max_used_memory AS avg_query_max_used_memory,
+	rs.max_duration as [max_duration],
+	rs.avg_duration as avg_duration,
+	rs.max_num_physical_io_reads as max_num_physical_io_reads,
+	rs.avg_num_physical_io_reads as avg_num_physical_io_reads,
+	rs.max_log_bytes_used as [max_log_bytes_used],
+	rs.avg_log_bytes_used as avg_log_bytes_used
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
+	JOIN sys.query_store_runtime_stats rs WITH (NOLOCK) ON p.plan_id = rs.plan_id
+	JOIN sys.query_store_runtime_stats_interval rsi WITH (NOLOCK) ON rs.runtime_stats_interval_id = rsi.runtime_stats_interval_id
+WHERE rsi.start_time >= @currIntervalStartDate AND rsi.end_time <= @currIntervalEndDate;
```

Comment:
Wouldn't it reduce amount of data (specially for MI) to collect aggregated data by query_id / plan_id even if you bucket say in 1 hour intervals - right now if someone changes collection interval to 5 mins or 10 mins will be a lot of data being collected.

Reply:
If the aggregation interval in SQL is set to 1 hour, the telegraf will return data only once per hour, whatever the collection interval is set to.

I did add aggregation by plan_id, execution_type and runtime_stats_interval_id, as suggested here:
https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-query-store-runtime-stats-transact-sql?view=sql-server-ver15 
-- to guarantee that there are no multiple records for the same time interval and plan_id.